mockingbird/src/commonMain/kotlin/com/careem/mockingbird/test/Capture.kt

```diff
@@ -47,18 +47,58 @@ public fun <T> capture(list: CapturedList<T>): CapturedMatcher<T> {
  */
 
 public class Slot<T> : Captureable {
+
+    private val genericSlot : GenericSlot<T> = initializeSlot()
+
+    public val captured: T?
+        get() {
+            return genericSlot.value
+        }
+
+    override fun storeCapturedValue(value: Any?) {
+        genericSlot.storeCapturedValue(value)
+    }
+}
+
+private fun <T> initializeSlot(): GenericSlot<T> {
```

Comment:
why do not expose this as public api as we discussed? like `fun <T> slot() : Slot<T>` same for list? and mark the `Slot` constructor as deprecated?